roles/node_prep/tasks/10_validation.yml

```diff
@@ -536,12 +536,13 @@
     - always
     - validation
 
-- name: Fail when bootstrapProvisioningIP are not set when virtualmedia option is enabled
+- name: Fail when bootstrapExternalStatic* variables are not set when virtualmedia option and static bootstrap booleans are enabled
   fail:
-    msg: "bootstrapProvisioningIP must be set when enable_virtualmedia is set"
+    msg: "bootstrapExternalStatic* variables must be set when enable_virtualmedia and static_bootstrap_extnet are set"
   when:
     - enable_virtualmedia|bool
-    - bootstrapProvisioningIP is undefined
+    - static_bootstrap_extnet|bool
+    - bootstrapExternalStaticIP is undefined or bootstrapExternalStaticGateway is undefined or bootstrapExternalStaticDNS is undefined
```

Comment:
bootstrapExternalStaticDNS was introduced in 4.14, you may not be able to assert its presence across the board

Reply:
Thanks for the clarification, I separated the conditions to isolate `bootstrapExternalStaticDNS` and validate OCP release is 4.14 and above.

Reply:
@derekhiggins 

I launched a test in 4.12 and because we cannot define `bootstrapExternalStaticDNS` when defining `bootstrapExternalStatic[IP/GW]` the bootstrapVM it's not able to resolve, since it's not using the correct DNS. By default it picked up the same IP as the GW, but that is not our DNS server. 
```
[kni@provisioner.cluster4.dfwt5g.lab 416-bug]$ ssh core@192.168.42.28
Warning: Permanently added '192.168.42.28' (ECDSA) to the list of known hosts.
Red Hat Enterprise Linux CoreOS 412.86.202402272018-0
  Part of OpenShift 4.12, RHCOS is a Kubernetes native operating system
  managed by the Machine Config Operator (`clusteroperator/machine-config`).

WARNING: Direct SSH access to machines is not recommended; instead,
make configuration changes via `machineconfig` objects:
  https://docs.openshift.com/container-platform/4.12/architecture/architecture-rhcos.html

---
This is the bootstrap node; it will be destroyed when the master is fully up.

The primary services are release-image.service followed by bootkube.service. To watch their status, run e.g.

  journalctl -b -f -u release-image.service -u bootkube.service
[core@localhost ~]$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 52:54:00:00:00:42 brd ff:ff:ff:ff:ff:ff
    inet 192.168.42.28/24 brd 192.168.42.255 scope global noprefixroute ens3
       valid_lft forever preferred_lft forever
    inet6 fd1c:61fe:bdf1:42::1c/128 scope global dynamic noprefixroute 
       valid_lft 40111sec preferred_lft 40111sec
    inet6 fe80::d87d:7023:e98b:410e/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
[core@localhost ~]$ cat /etc/resolv.conf 
# Generated by NetworkManager
nameserver 127.0.0.1
nameserver 192.168.42.1
[core@localhost ~]$ ip r
default via 192.168.42.1 dev ens3 proto static metric 100 
192.168.42.0/24 dev ens3 proto kernel scope link src 192.168.42.28 metric 100 
[core@localhost ~]$ host
host         hostid       hostname     hostnamectl  
[core@localhost ~]$ host google.fr
;; connection timed out; no servers could be reached

[core@localhost ~]$ sudo journalctl -f
-- Logs begin at Tue 2024-09-10 15:21:36 UTC. --
Sep 10 21:15:15 localhost.localdomain release-image-download.sh[4691]: time="2024-09-10T21:15:15Z" level=warning msg="Failed, retrying in 1s ... (2/3). Error: initializing source docker://registry.ci.openshift.org/ocp/release@sha256:cfef48744df1180b97eaa688c0b84942c997b92fea0bf84ba20b83df39cc4e76: (Mirrors also failed: [registry.dfwt5g.lab:4443/ocp-4.12/4.12.0-0.nightly-2024-09-04-103217@sha256:cfef48744df1180b97eaa688c0b84942c997b92fea0bf84ba20b83df39cc4e76: pinging container registry registry.dfwt5g.lab:4443: Get \"https://registry.dfwt5g.lab:4443/v2/\": dial tcp: lookup registry.dfwt5g.lab on 192.168.42.1:53: read udp 192.168.42.28:41065->192.168.42.1:53: i/o timeout]): registry.ci.openshift.org/ocp/release@sha256:cfef48744df1180b97eaa688c0b84942c997b92fea0bf84ba20b83df39cc4e76
```
This probably works when the GW IP is the same as the DNS, then I think we should only allow to set `bootstrapExternalStatic[IP/GW]` in 4.14 deployments when we can define bootstrapExternalStaticDNS as well. Wdyt?

Reply:
You are correct, in 4.12 ipi assumes that the GW IP is the same as the DNS, I'm thinking its better to allow the absence of bootstrapExternalStaticDNS when <4.14, knowing the test env doesn't support this scenario. Its not possible to use it in the test env here but may be valid for other users.